code/modules/mob/living/carbon/species.dm

```diff
@@ -202,12 +195,16 @@
 			if(SA_pp > SA_para_min) // Enough to make us paralysed for a bit
 				H.Paralyse(3) // 3 gives them one second to wake up and run away a bit!
 				if(SA_pp > SA_sleep_min) // Enough to make us sleep as well
-					H.sleeping = min(H.sleeping+2, 10)
-			else if(SA_pp > 0.15)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
+					H.sleeping = max(H.sleeping+2, 10)
+			else if(SA_pp > 0.01)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
 				if(prob(20))
 					spawn(0) H.emote(pick("giggle", "laugh"))
-			SA.moles = 0
 
+	handle_temperature(breath, H)
+
+	return 1
+
+/datum/species/proc/handle_temperature(datum/gas_mixture/breath, var/mob/living/carbon/human/H) // called by human/life, handles temperatures
 	if( (abs(310.15 - breath.temperature) > 50) && !(RESIST_HEAT in H.mutations)) // Hot air hurts :(
```

Comment:
Not sure if this should be 310.15 or species.body_temperature; doesn't really impact anyone other than Diona, so I'm going to guess this should be left alone.